[Frontend] Added Validation in the Add FAQ Form

mrunalvilas opened the pull request:
## [Frontend] Add Validation in the Add FAQ Form

 Closes: #538

## Proposed changes:
Added Validations to the Add FAQ form from Admin Dashboard.

### Brief description of what is fixed or changed
1. Validation that checks if the question and answer fileds are empty or not.
2. Ensure that the length of the question and answer fields is 8.

## Types of changes

_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes do not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of a change in the user interface
1.
![22](https://user-images.githubusercontent.com/60320511/111884048-ffacde00-89e4-11eb-9ffb-6e616feaa7ce.PNG)



2.
![screencapture-localhost-3000-dashboard-2021-03-20-21_08_24](https://user-images.githubusercontent.com/60320511/111883981-93ca7580-89e4-11eb-8405-af039052b034.png)

Kajol-Kumari commented on frontend/src/pages/Admin/Components/Faq/AddFaq/add-faq.module.scss:
```diff
@@ -12,6 +12,12 @@
+  margin-top: 0;
+}
+
+
```
please leave only single line gap here

Kajol-Kumari reviewed (changes requested):
Hey @mrunalvilas As we are using `joi-validator` for validation purpose. So please implement the validation here as well using the same inorder to be consistent. Also the validation doesn't seem to be working, please check it once on your end

mrunalvilas commented:
https://user-images.githubusercontent.com/60320511/111899152-b72b0a00-8a50-11eb-8c69-b6f4dd015bae.mp4

here is a working video of the applied changes
1. I have implemented a similar validation function as is used in the sign-up field from the footer.
I would replace it with joi-validator (just want some reference in which field is it used )
2. I have replaced 2 line spaces with one line space in the add-faq.module.css file.

Kajol-Kumari commented:
Hey @mrunalvilas We need the validation's error message to appear just below the respective input fields(just like we are doing it for other forms). Also as we don't want the default type error messages to appear anymore, so please remove them

Kajol-Kumari commented:
> Hey @mrunalvilas We need the validation's error message to appear just below the respective input fields(just like we are doing it for other forms). Also as we don't want the default type error messages to appear anymore, so please remove them

Hey @mrunalvilas can you please provide an update on this?

mrunalvilas commented:
Yes,  will update shortly

Kajol-Kumari reviewed (changes requested):
Hey @mrunalvilas can you please fix the padding of the error msg. It should appear near to the input box (Please see the other forms for reference)

Rest looks good to go :)

Kajol-Kumari reviewed (approved):
Thanks for the quick fix @mrunalvilas 
This looks good to go now :)
